src/sinks/splunk_hec/metrics/encoder.rs

```diff
@@ -55,8 +55,8 @@ impl HecMetricsEncoder {
             .into_iter()
             .flatten()
             // skip the metric tags used for templating
-            .filter(|(k, _)| !metadata.templated_field_keys.contains(k))
-            .map(|(k, v)| (k.as_str(), HecFieldValue::Str(v.as_str())))
+            .filter(|(k, _)| !metadata.templated_field_keys.iter().any(|f| f == k))
```

Comment:
Could this not use `MetricTags::contains_key`?

Reply:
I'm not seeing how `MetricTags::contains_key` could be used here. What we want is a copy of all the tags that are not in `templated_field_keys`, which is type `Vec<String>`.

Reply:
Ah derp, yeah... I was thinking it was doing something `contains_key`-esque, but now I see it was just doing the same exact thing as you have here. Ignore me. :)